### DISCUSSION TITLE
Add the ability for case insensitive URLs

### DISCUSSION THREAD

**Discussion by maxvaser** [laravel/framework#38188] [Ideas]
Please add for the ability to make URLs case insensitive. This would be very helpful for a lot of reasons and especially for vanity/shortcut URLs  where someone may type in the address bar and capitalize a portion of it. I haven't been able to find a work around or solution that works

Like:
- https://myapp.com/Promo
- https://myapp.com/PROMO
- https://myapp.com/promo

**Comment by dennisprudlo** [comment]
I don't think Laravel has something for that out of the box. If you want that just for this promo use case I'd recommend to use a parameter and a pattern.

https://laravel.com/docs/8.x/routing#parameters-regular-expression-constraints
